### PR TITLE
Gives the Bridge Assistant Axe Delusions

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request
The Bridge Assistant can wield items now! You can use defibs, spears, whatever!
Instead you now have a brain trauma about your duty to the Bridge fire axe. You are only really happy when it's in its cabinet, but you are especially unhappy if it's in the hands of someone who isn't in command staff, or laying somewhere outside of the bridge, or not existing at all (now that one has you fucked up).
When you wield it yourself, it also makes you unhappy, and it whispers to you.
image
image
image

## Why It's Good For The Game
It's pretty funny and the original method wasn't great. Being too weak to wield something doesn't make sense because if you're using two hands you're stronger. It prevented you from using stuff like defibs as well.

## Changelog
:cl:
add: The Bridge Assistant's no item wielding disability was replaced with a duty-sworn axe protection brain trauma.
/:cl:

### CODE REVIEW

File: code/datums/brain_damage/special.dm [hunk 1]
```diff
@@ -525,3 +525,173 @@
...
+		if(istype(get_axe_location(), /area/station/command) && istype(GLOB.bridge_axe.loc, /obj/structure/fireaxecabinet))
+			owner.add_mood_event("fireaxe", /datum/mood_event/axe_cabinet)
+		else if(owner.is_holding(GLOB.bridge_axe))
+			on_equip(owner, GLOB.bridge_axe)
+		else
+			owner.add_mood_event("fireaxe", /datum/mood_event/axe_neutral)
+		RegisterSignal(owner, COMSIG_MOB_EQUIPPED_ITEM, PROC_REF(on_equip))
+		RegisterSignal(owner, COMSIG_MOB_UNEQUIPPED_ITEM, PROC_REF(on_unequip))
+		RegisterSignal(owner, COMSIG_MOB_EXAMINING, PROC_REF(on_examine))
+		RegisterSignal(GLOB.bridge_axe, COMSIG_ITEM_AFTERATTACK, PROC_REF(on_axe_attack))
```
Comment (SmArtKar) [inline code/datums/brain_damage/special.dm:615]: early returns please

File: code/datums/brain_damage/special.dm [hunk 2]
```diff
@@ -525,3 +525,173 @@
...
+		return
+	owner.set_jitter_if_lower(3 SECONDS)
+	if(picked_up == GLOB.bridge_axe)
+		to_chat(owner, span_hypnophrase("I have it. It's time to put it back."))
+		owner.add_mood_event("fireaxe", /datum/mood_event/axe_held)
+	else
+		ADD_TRAIT(picked_up, TRAIT_NODROP, type)
+		to_chat(owner, span_warning("...This is not the one I'm looking after."))
+		owner.Immobilize(2 SECONDS)
+		addtimer(CALLBACK(src, PROC_REF(throw_faker), picked_up), 2 SECONDS)
```
Comment (SmArtKar) [inline code/datums/brain_damage/special.dm:643]: early returns

File: code/datums/brain_damage/special.dm [hunk 3]
```diff
@@ -525,3 +525,173 @@
...
+		return
+	if(istype(new_location, /obj/structure/fireaxecabinet))
+		if(istype(get_area(new_location), /area/station/command))
+			to_chat(owner, span_nicegreen("Ah! Back where it belongs!"))
+			owner.add_mood_event("fireaxe", /datum/mood_event/axe_cabinet)
+			INVOKE_ASYNC(owner, TYPE_PROC_REF(/mob, emote), "smile")
+		else
+			to_chat(owner, span_warning("Leaving it outside of command? Am I sure about that?"))
+			owner.add_mood_event("fireaxe", /datum/mood_event/axe_neutral)
+		return
```
Comment (SmArtKar) [inline code/datums/brain_damage/special.dm:669]: early returns

**Review by SmArtKar — CHANGES_REQUESTED** [review]

**Comment by mc-oofert** [comment]
moodlets stop literally nobody

**Comment by Fikou** [comment]
i only played webber in dst 7 years ago

**Review by SmArtKar — APPROVED** [review]

**Comment by SmArtKar** [comment]
> moodlets stop literally nobody

consider this: its funny

**Comment by moocowswag** [comment]
Won't bridge assistant get a negative moodlet if someone who was promoted to command gets the axe, giving them a reason to escalate against them?

**Comment by Fikou** [comment]
> Won't bridge assistant get a negative moodlet if someone who was promoted to command gets the axe, giving them a reason to escalate against them?

you never truly trust the other mfers...

**Comment by moocowswag** [comment]
Works for me

**Comment by btcbuster** [comment]
So if the axe somehow gets out of bridge the assistant has to stop doing the rest of their job to go get it back.... even if that person is authorized to have it?  It's just kinda weird

**Comment by MrMelbert** [comment]
It's very funny how any time negative moodlets are used to deter behavior, 50% of the time you see "negative moodlets are not that bad, no one cares" and the other 50% of the time you see "negative moodlets are horrible to play with and too disruptive"

File: code/game/objects/items/fireaxe.dm [hunk 4]
```diff
@@ -34,6 +36,9 @@
 
 /obj/item/fireaxe/Initialize(mapload)
 	. = ..()
+	if(!GLOB.bridge_axe && istype(get_area(src), /area/station/command))
+		GLOB.bridge_axe = src
```
Comment (MrMelbert) [inline code/game/objects/items/fireaxe.dm:40]: Need to `null` in destroy

Comment (Fikou) [reply code/game/objects/items/fireaxe.dm:40]: that does track

**Comment by Dorsisdwarf** [comment]
> So if the axe somehow gets out of bridge the assistant has to stop doing the rest of their job to go get it back.... even if that person is authorized to have it? It's just kinda weird

Keeping the axe out of the hands of random shitheads is more important than anything else they could be doing tbh

**Comment by Thunder12345** [comment]
> It's very funny how any time negative moodlets are used to deter behavior, 50% of the time you see "negative moodlets are not that bad, no one cares" and the other 50% of the time you see "negative moodlets are horrible to play with and too disruptive"

That just tells you which people know how to farm positive moodlets and which don't

**Comment by DaCoolBoss** [comment]
Could the trauma be turned off if there's no fireaxe in the map's bridge? Current mapping convention is that there's always one there, but like any convention that might change in future. Also check that no weirdness happens if there's two axes mapped into the bridge. Looking at the code, both will be "the bridge axe" with associated moodlets and features.

Also I feel like the penalties for a missing fireaxe should either be on a delay (30sec+) or require seeing the axe case. Possibly both, with whatever coming first triggering the moodlet. Instant remote indicators that someone's committing a crime are no good.

**Comment by Fikou** [comment]
> Could the trauma be turned off if there's no fireaxe in the map's bridge? Current mapping convention is that there's always one there, but like any convention that might change in future. Also check that no weirdness happens if there's two axes mapped into the bridge. Looking at the code, both will be "the bridge axe" with associated moodlets and features.
> 
> Also I feel like the penalties for a missing fireaxe should either be on a delay (30sec+) or require seeing the axe case. Possibly both, with whatever coming first triggering the moodlet. Instant remote indicators that someone's committing a crime are no good.

there can only be one GLOB.bridge_axe, and its not instant as it is a random chance of getting a bad feeling

**Comment by moocowswag** [comment]
Bridge assistant on the escape shuttle on the way to centcomm when someone that got left behind decides to tide the bridge and steal the fire axe.

image

**Comment by EricZilla** [comment]
It would be funny if they go a negative moodlet for the axe being on the shuttle too, something like "A true leader goes down with the ship"

**Comment by KingkumaArt** [comment]
> and it whispers to you. 
https://images-ext-1.discordapp.net/external/kNQBaz0JpIpLXTJg3Xlmmzk6lshcrFiFSjtAewIDG7k/https/media.tenor.com/xD_oGeogfM4AAAPo/green-goblin.mp4

**Comment by Loafin34** [comment]
okay this is good but what if the fire axe cabinet gets destroyed. can you build a new one and put it in there just fine?

**Comment by Fikou** [comment]
> okay this is good but what if the fire axe cabinet gets destroyed. can you build a new one and put it in there just fine?

yes

**Comment by moocowswag** [comment]
What about labeling other rooms as command with blueprints

**Comment by Fikou** [comment]
shut up bro

File: code/datums/brain_damage/special.dm [hunk 5]
```diff
@@ -525,3 +525,175 @@
...
+	else
+		examine_strings += span_warning("It's a simulacra, a fake axe made to fool the masses.")
+
+/datum/brain_trauma/special/axedoration/proc/on_axe_attack(obj/item/axe, atom/target, mob/user, click_parameters)
+	SIGNAL_HANDLER
+	if(user != owner)
+		return
+	talk_tuah(pick(hurt_lines))
+
+/datum/brain_trauma/special/axedoration/proc/talk_tuah(sent_message = "Hello World.")
```
Comment (nikothedude) [inline code/datums/brain_damage/special.dm:690]: i love talk tuah

Comment (Fikou) [reply code/datums/brain_damage/special.dm:690]: chop on that thang

**Review by SmArtKar — CHANGES_REQUESTED** [review]

**Comment by moocowswag** [comment]
Can bridge assistant roll antag

**Comment by Fikou** [comment]
they shouldnt but theres a bug where they could idk if its fixed

**Comment by Hotmud9** [comment]
Oh god what if someone just built a bunch of empty fire axe cabinets in the bridge JUST to fuck with the bridge assistant

**Pull request merged**